Source/Urho3D/Resource/XMLElement.cpp

```diff
@@ -439,17 +439,16 @@ bool XMLElement::SetVariantValue(const Variant& value)
 
     case VAR_CUSTOM_STACK:
     {
-        if (const auto* ptr = value.GetCustomPtr<SharedPtr<Serializable>>())
+        if (const Serializable* object = value.GetCustom<SharedPtr<Serializable>>())
         {
-            if (ptr->NotNull())
+            SetAttribute("type", object->GetTypeName());
+            if (!object->SaveXML(*this))
             {
-                Serializable* object = ptr->Get();
-                SetAttribute("type", object->GetTypeName());
-                if (!object->SaveXML(*this))
-                    RemoveChildren();
-                else
-                    return true;
+                RemoveAttribute("type");
+                RemoveChildren();
```

Comment:
Don't want to return false here?

Reply:
It falls through and returns `false` at https://github.com/rokups/rbfx/blob/d06174b1926cdac028338ab4ec40d2dc75dd8dc0/Source/Urho3D/Resource/XMLElement.cpp#L455